parser/prism/Translator.cc

```diff
@@ -0,0 +1,1059 @@
+#include "Translator.h"
+
+template class std::unique_ptr<sorbet::parser::Node>;
+
+using std::make_unique;
+using std::unique_ptr;
+
+namespace sorbet::parser::Prism {
+
+// Indicates that a particular code path should never be reached, with an explanation of why.
+// Throws a `sorbet::SorbetException` in debug mode, and is undefined behaviour otherwise.
+template <typename... TArgs>
+[[noreturn]] void unreachable(fmt::format_string<TArgs...> reason_format_str, TArgs &&...args) {
+    if constexpr (sorbet::debug_mode) {
+        Exception::raise(reason_format_str, std::forward<TArgs>(args)...);
+    } else {
+        // Basically a backport of C++23's `std::unreachable()`:
+        // > `ABSL_UNREACHABLE()` is an unreachable statement.  A program which reaches
+        // > one has undefined behavior, and the compiler may optimize accordingly.
+        ABSL_UNREACHABLE();
+    }
+}
```

Comment:
It's not clear to me what this is method is needed for. For any current call to `unreachable`, couldn't you replace it with `ENFORCE` if you don't want to raise in non-debug builds?

Taking a step back, is there a reason why we don't want to raise in non-debug builds? Why not just call `Exception::raise`?